api/v1alpha1/oidc_types.go

```diff
@@ -153,6 +159,27 @@ type OIDCProvider struct {
 	TokenEndpoint *string `json:"tokenEndpoint,omitempty"`
 }
 
+// OIDCDenyRedirect defines headers to match against the request to deny redirect to the OIDC Provider.
+// +notImplementedHide
+type OIDCDenyRedirect struct {
+	// Defines the headers to match against the request to deny redirect to the OIDC Provider.
+	Headers []OIDCDenyRedirectHeader `json:"headers"`
+}
+
+// OIDCDenyRedirectHeader defines how a header is matched
+// +notImplementedHide
+type OIDCDenyRedirectHeader struct {
+	// Specifies the name of the header in the request.
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+	// The type of match to perform on the header.
+	// Valid MatchType values are "Exact", "Prefix", "Suffix", "RegularExpression".
+	// +kubebuilder:validation:Enum=Exact;Prefix;Suffix;RegularExpression
+	Type *StringMatchType `json:"type"`
```

Comment:
instead of redefining these field you can achieve the same by defining it as
```
Name string `json:"name"`
StringMatch
```

Reply:
aahh 

```go
type OIDCDenyRedirectHeader struct {
	// Specifies the name of the header in the request.
	// +kubebuilder:validation:MinLength=1
	Name        string `json:"name"`
	StringMatch `json:",inline"`
}
```

does the trick